src/graphiql/GraphiQlExtensions.cs

```diff
@@ -14,18 +14,17 @@ public static class GraphiQlExtensions
         public static IApplicationBuilder UseGraphiQl(this IApplicationBuilder app)
             => UseGraphiQl(app, DefaultPath);
 
-        public static IApplicationBuilder UseGraphiQl(this IApplicationBuilder app, string path)
-            => UseGraphiQlIml(app, path);
-        
-        private static IApplicationBuilder UseGraphiQlIml(this IApplicationBuilder app, string path)
+        /// <param name="apiPath">In some scenarios it makes sense to specify the API path and file server path independently
+        /// Examples: hosting in IIS in a virtual application (myapp.com/1.0/...) or hosting API and documentation separately</param>
+        public static IApplicationBuilder UseGraphiQl(this IApplicationBuilder app, string fileServerPath, string apiPath = null)
```

Comment:
Could we create an overload of this method so we don't have to use an optional argument?

So it looks more like this:

```csharp
public static IApplicationBuilder UseGraphiQl(this IApplicationBuilder app, string fileServerPath)
    => UseGraphiQl(app, fileServerPath, null);

public static IApplicationBuilder UseGraphiQl(this IApplicationBuilder app, string fileServerPath, string apiPath);
```

Also, to help reduce breaking changes and a major version release could we create the original method signature but hide it from the IDE, this would mean anyone using a named parameter for path wouldn't break?
```csharp

[System.ComponentModel.EditorBrowsable(System.ComponentModel.EditorBrowsableState.Never)]
public static IApplicationBuilder UseGraphiQl(this IApplicationBuilder app, string path)
```